src/main/java/TARS/ui/TARS.java

```diff
@@ -0,0 +1,559 @@
+
+                if (commandType == CommandType.BYE) {
+                    break;
+                }
+
+                String commandBody = parseCommandBody(line, commandType);
+                commandHandler(commandType, commandBody);
+                writeFile();
+
+            } catch (Exception e) {
+                printResponseMessage(e.getMessage());
+            }
+
+        }
+
+        printGoodbyeMessage();
+        sc.close();
+    }
+
+    private static void printResponseMessage(String message) {
+        System.out.println(LINE_SEPERATOR);
+        System.out.println(message);
+        System.out.println(LINE_SEPERATOR);
+    }
+
+    /**
+     * Prints the hello message to welcome the user.
+     */
+    private static void printHelloMessage() {
+        printResponseMessage(HELLO_MESSAGE);
+    }
+
+    /**
+     * Prints the goodbye message before exiting.
+     */
+    private static void printGoodbyeMessage() {
+        printResponseMessage(GOODBYE_MESSAGE);
+    }
+
+    /**
+     * Parses user input line for commandType
+     * Validation throws exception for empty strings, returns TARS.command.CommandType.INVALID for unknown commands
+     * @param line user input String as a command
+     * @return first word in line as commandType
+     */
+    private static CommandType parseCommandType(String line) throws IndexOutOfBoundsException, IllegalArgumentException {
+        // Check for empty line
+        line = line.trim();
+        if (line.isEmpty()) {
+            throw new IndexOutOfBoundsException("Invalid Command: . Command cannot be empty. Use 'help' for valid commands.");
+        }
+
+        // Parse first word for TARS.command.CommandType
+        int commandEndIndex = line.indexOf(' ');
+        if (commandEndIndex == -1) {
+            commandEndIndex = line.length(); // case of single word commands e.g. list with no spaces
+        }
+
+        String commandString = line.substring(0, commandEndIndex);
+        return CommandType.fromString(commandString);
+    }
+
+
+    /**
+
+     * parses commandBody for arguments from the user input after extracting command type
+     * Validates the format of command body based on command type
+     * Validation Rules
+     * LIST: must not contain any arguments
+     * MARK / UNMARK: must be a valid 1-indexed integer representing task index.
+     * TODO: arguments must not contain any flags
+     * DEADLINE: must only contain /by flag followed by due date
+     * EVENT: must only contain /from and /to flags in order, each followed by start and end times
+     * @param line full user input String
+     * @param commandType enum TARS.command.CommandType extracted from user input String
+     * @return validated commandBody String
+     * @throws TARSInvalidCommandBodyException if the commandBody does not meet validation criteria
+     */
+    private static String parseCommandBody(String line, CommandType commandType)  throws TARSInvalidCommandBodyException {
```

Comment:
This function is way over >30 LOC. Consider refactoring the internal levels of code into separate methods.